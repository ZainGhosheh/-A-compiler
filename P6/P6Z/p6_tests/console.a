main : () -> void {
  a:int;
  b:bool;

  fromconsole a;
  fromconsole b;

  toconsole "string";
  toconsole b or true;
  toconsole a + 1203;
}