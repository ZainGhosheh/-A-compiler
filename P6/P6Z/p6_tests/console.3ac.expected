[BEGIN GLOBALS]
main
str_0 "string"
[END GLOBALS]
[BEGIN <init> LOCALS]
[END <init> LOCALS]
fun_<init>: enter <init>
lbl_0:      leave <init>
[BEGIN main LOCALS]
a (local var of 8 bytes)
b (local var of 8 bytes)
tmp0 (tmp var of 8 bytes)
tmp1 (tmp var of 8 bytes)
[END main LOCALS]
main:       enter main
            READ [a]
            READ [b]
            WRITE str_0
            [tmp0] := [b] OR64 1
            WRITE [tmp0]
            [tmp1] := [a] ADD64 1203
            WRITE [tmp1]
lbl_1:      leave main